### PR TITLE
[SAR-10030] Upgrade to hmrc-mongo

### PR DESCRIPTION
**Pull request by refaque-ahmad**

### CODE REVIEW

File: app/uk/gov/hmrc/soletraderidentification/controllers/JourneyDataController.scala [hunk 1]
```diff
@@ -99,7 +103,7 @@ class JourneyDataController @Inject()(cc: ControllerComponents,
             _ => NoContent
```
Comment (woollett-r) [inline app/uk/gov/hmrc/soletraderidentification/controllers/JourneyDataController.scala]: You could update this to throw an error if the repository method returns "false". This basically means there is no journey with the id provided. Have a look at how the existing code behaves. I think the method will execute without error if the journey id is ok, but there is no matching data field. However, if there is no journey data an error will be raised.

File: app/uk/gov/hmrc/soletraderidentification/controllers/JourneyDataController.scala [hunk 2]
```diff
@@ -111,7 +115,7 @@ class JourneyDataController @Inject()(cc: ControllerComponents,
...
```
Comment (woollett-r) [inline app/uk/gov/hmrc/soletraderidentification/controllers/JourneyDataController.scala]: Should you throw an error if the "removeJourneyData" method returns false?

File: app/uk/gov/hmrc/soletraderidentification/repositories/JourneyDataRepository.scala [hunk 3]
```diff
@@ -16,120 +16,101 @@
...
-    name = Some(TtlIndexName),
-    options = BSONDocument("expireAfterSeconds" -> appConfig.timeToLiveSeconds)
-  )
-
-  private def setIndex(): Unit = {
-    collection.indexesManager.drop(TtlIndexName) onComplete {
-      _ => collection.indexesManager.ensure(ttlIndex)
+      )
+    ).toFuture.map {
+      _ != null
```
Comment (woollett-r) [inline app/uk/gov/hmrc/soletraderidentification/repositories/JourneyDataRepository.scala:90]: By default "findOneAndReplace" returns the original document not the replacement. Maybe we should make the method return the replacement. One for the team to discuss maybe.

File: it/controllers/JourneyDataControllerISpec.scala [hunk 4]
```diff
@@ -298,49 +304,28 @@ class JourneyDataControllerISpec extends ComponentSpecHelper with CustomMatchers
...
-        val res = delete(s"/journey/$testJourneyId")
+      insertById(testJourneyId, testInternalId, Json.obj(testDataKey -> testDataValue, secondTestDataKey -> secondTestDataValue))
 
-        res.status mustBe INTERNAL_SERVER_ERROR
+      val res = delete(s"/journey/$testJourneyId")
 
-        findById(testJourneyId) mustBe None
-      }
-    }
-    "the user cannot be authorised" should {
```
Comment (woollett-r) [inline it/controllers/JourneyDataControllerISpec.scala:336]: Should this test have been removed?

File: it/utils/JourneyDataMongoHelper.scala [hunk 5]
```diff
@@ -20,24 +20,24 @@ import org.scalatest.{BeforeAndAfterEach, Suite}
...
+    await(repo.getJourneyData(journeyId, authInternalId))
 
   def insertById(journeyId: String, internalId: String, jsonData: JsObject = Json.obj()): Unit =
-    await(repo.collection.insert(true).one(Json.obj("_id" -> journeyId, "authInternalId" -> internalId) ++ jsonData))
+    await(repo.collection.insertOne(
+      Json.obj(
+        "_id" -> journeyId,
+        "authInternalId" -> internalId
+      ) ++ jsonData
+    ).toFuture())
```
Comment (woollett-r) [inline it/utils/JourneyDataMongoHelper.scala]: The method insertOne returns an instance of SingleObservable. To really match the declared return type you could use ".toFuture().map(_ => Unit)".

File: app/uk/gov/hmrc/soletraderidentification/controllers/JourneyDataController.scala (hunk 2 above)
Comment (refaque-ahmad) [reply app/uk/gov/hmrc/soletraderidentification/controllers/JourneyDataController.scala]: added

File: it/controllers/JourneyDataControllerISpec.scala (hunk 4 above)
Comment (refaque-ahmad) [reply it/controllers/JourneyDataControllerISpec.scala:336]: added

File: it/utils/JourneyDataMongoHelper.scala (hunk 5 above)
Comment (refaque-ahmad) [reply it/utils/JourneyDataMongoHelper.scala]: done

File: app/uk/gov/hmrc/soletraderidentification/repositories/JourneyDataRepository.scala (hunk 3 above)
Comment (refaque-ahmad) [reply app/uk/gov/hmrc/soletraderidentification/repositories/JourneyDataRepository.scala:90]: Yeah makes sense to discuss with the team

**Review by woollett-r — APPROVED** [review]

**Pull request merged**